File: 30 Days of Code/Day 10 Binary Numbers/Solution.cs
public class Solution {
    private static string ConvertToBinary( int n ) {
 
         if( n == 0 ) {
             return "0";
         }
         if( n == 1 ) {
             return "1";
         }
 
        return $"{ConvertToBinary( n / 2 )}{n % 2}";
     }
 
    private static string IntToBinary( int n ) {
        const int mask = 1;
        string binary = String.Empty;

        while( number > 0 ) {
            binary = ( number & mask ) + binary;
            number = number >> 1;
        }

        return binary;
    }

    private static int GetMaxConsecutive( string binary ) {
        int max = 0;
        int count = 0;

        foreach( var digit in binary ) {
            if( digit == '0' ) {
                if( count >  max ) {
                    max = count;
                }
                count = 0;
            } else {
                count += 1;
            }
        }

        if( count > max ) {
            max = count;
        }
        
        return max;
    }


    public static void Main( String[ ] args ) {
        int n = Convert.ToInt32( Console.ReadLine( ) );
        string binary = ConvertToBinary( n );
        int max = GetMaxConsecutive( binary );
 
        Console.WriteLine( max );
    }
 }

File: 30 Days of Code/Day 10 Binary Numbers/Solution.py
import sys

def convertToBinary( number ):
    if number == 0:
        return "0"
    if number == 1:
        return "1"
    return convertToBinary( int( number / 2 ) ) + str( number % 2 )


def intToBinary( number ):
    MASK = 1
    binary = ""

    while number > 0:
        binary = str( number & MASK ) + binary
        number = number >> 1
    return binary


def getMaxConsecutive( binary ):
    max = 0
    count = 0
    for digit in binary:
        if digit == '1':
            count += 1
        else: 
            if count > max:
                max = count
            count = 0
    if count > max:
        max = count
    return max


n = int( input( ).strip( ) )
binary = intToBinary( n )
result = getMaxConsecutive( binary )
print( result )